### PR TITLE
Refactor: Bring `Inter` and `IBM Plex Mono` Fonts to Prod / Update Implementation

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
## Description

This PR does the following for the `Inter` font:
- Bring the font to prod via theme config
- Remove the `Fonts` component in favor of a static CSS file of the font faces
  - Stores this new file in the static `fonts` folder
- Expose the `static` directory in Storybook to allow visibility of the Inter font

Also adds the `IBM Plex Mono` font in regular weight for the `monospace` font family stack

### CODE REVIEW

**Comment by TylerAPfledderer** [comment]
@pettinarip hmmm Chromatic is failing because the Chakra CLI can not locate the theme file. This is due to that CSS import I applied. Need to inquire with the Chakra team on this.

**Comment by nloureiro** [comment]
As far as I could text, the new Inter font works really well!!! It improves a lot the readability.

One question:
Did you remove the monospace font family?
it's where I see the main differences. Very small but changes a bit
Screen Shot 2023-05-09 12 29 13 AM

We could keep the same monospace font family to avoid loading another font.
If you don't see that as a more significant load, I can also work on a new monospace font.

@TylerAPfledderer @pettinarip , what do you think?

**Comment by nloureiro** [comment]
As far as I could text, the new Inter font works really well!!! It improves a lot the readability.

One question:
Did you remove the monospace font family?
it's where I see the main differences. Very small but changes a bit
Screen Shot 2023-05-09 12 29 13 AM

We could keep the same monospace font family to avoid loading another font.
If you don't see that as a more significant load, I can also work on a new monospace font.

@TylerAPfledderer @pettinarip , what do you think?

**Comment by TylerAPfledderer** [comment]
@nloureiro correct, we would want to ensure a font is available in the project here for build.

If I'm not mistaken @pettinarip adding a font file locally should only add somewhere less than a kilobyte? If not way less!

**Comment by nloureiro** [comment]
> > If I'm not mistaken @pettinarip adding a font file locally should only add somewhere less than a kilobyte? If not way less!
> 
> Not sure if I follow you on that. It will depend on how many fonts we want to add and the size of each one of them.
> 
> https://github.com/IBM/plex/releases/tag/v6.3.0 this is the font we want to add, right? https://www.ibm.com/plex/
> 
> Questions @nloureiro:
> 
> * Do you think we need to support more than just latin characters?
> * Do you think we need to support more than the `regular` weight fonts?
> 
> If we are only going to support `regular-latin` chars, here are the fonts to include:
> 
> * IBMPlexMono-Regular-Latin1.woff2 - 16.3kb
> * IBMPlexMono-Regular-Latin2.woff2 - 13.3k
> * IBMPlexMono-Regular-Latin3.woff2 - 6.6kb
> 
> Note: not sure the difference between 1, 2, and 3. Perhaps we could just use the `Latin1` and that could be enough, not sure.


Just Regular is fine. We don´t use any other weight on monospace.
Let me see if we can find the difference in these charsets, but I would guess that Latin1 might be enough.

**Comment by pettinarip** [comment]
Cool! so, lets wait for @nloureiro's confirmation on what font we need to load and we can add them @TylerAPfledderer

**Comment by pettinarip** [comment]
Sounds good @nloureiro!

Other parts where we are using the font are:
- Home page modals (for the code examples)
- Call to contribute widget. E.g. https://ethereum.org/en/developers/docs/smart-contracts/composability/
- Stats on the L2 page https://ethereum.org/en/layer-2/
- Titles for the tutorials, learn by coding section, the docs. E.g. https://ethereum.org/en/developers/tutorials/eip-1271-smart-contract-signatures/
image

**Comment by nloureiro** [comment]
> Sounds good @nloureiro!
> 
> Other parts where we are using the font are:
> 
> * Home page modals (for the code examples)
> * Call to contribute widget. E.g. https://ethereum.org/en/developers/docs/smart-contracts/composability/
> * Stats on the L2 page https://ethereum.org/en/layer-2/
> * Titles for the tutorials, learn by coding section, the docs. E.g. https://ethereum.org/en/developers/tutorials/eip-1271-smart-contract-signatures/
>   image


Interesting, all these sections that now have the monospace font, I already changed on the Design system.

that being said, I don´t see any change in these examples on the deploy preview, I assume we are using other Font family calls for these examples
Screenshot 2023-05-10 at 15 13 45

Can you check where we are using it?

**Comment by pettinarip** [comment]
Hmm on [the preview deploy for this PR](https://ethereumorgwebsitedev01-tylerapfleddererethereumo01219.gatsbyjs.io/en/layer-2/), I don't see the old font definitions, I see this:
image

Can you double-check that you are using the correct preview deploy link?

**Comment by nloureiro** [comment]
> @nloureiro hmm on [the preview deploy for this PR](https://ethereumorgwebsitedev01-tylerapfleddererethereumo01219.gatsbyjs.io/en/layer-2/), I don't see the old font definitions, I see this: image
> 
> Can you double-check that you are using the correct preview deploy link?

yes, it's on the preview deploy.

this is the lines that are supposed to be removed. That's why I think we declare the monospace font in other files
Screenshot 2023-05-10 at 23 22 44

**Comment by TylerAPfledderer** [comment]
@pettinarip @nloureiro can never forget that there is the old theme file that also contains the stack, and might (I guess only sometimes) rear its head! 😅

**Comment by nloureiro** [comment]
> @pettinarip @nloureiro can never forget that there is the old theme file that also contains the stack, and might (I guess only sometimes) rear its head! 😅

Right! That might be it!!!!

In this case, I would say that we can move forward in adding the IBM Plex mono as a monospace font.


I've added to the design system typography section
Screen Shot 2023-05-10 05 31 57 PM
Screen Shot 2023-05-11 04 04 35 PM

@pettinarip do you think it's better to do it on this PR? or create a new one on top of this one?

My take it's that once re touch the monospace family here, we might as well use this to add this font and QA all at the same time

**Comment by TylerAPfledderer** [comment]
@nloureiro @pettinarip I'll at least push a commit to revert the original stack back into the Chakra config

**Review by nloureiro — APPROVED** [review]

File: src/@chakra-ui/gatsby-plugin/theme.ts [hunk 1]
```diff
@@ -29,16 +29,6 @@ const config: ThemeConfig = {
...
   config,
-  // TODO: fonts object to be removed (it'll now be in the 'foundation' directory)
-  fonts: {
-    // old fonts from the previous theme
-    // TODO: update the fonts when we transition to the Design System
-    heading:
-      "system-ui, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
-    body: "system-ui, -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
-    monospace:
-      "SFMono-Regular, Consolas, 'Roboto Mono', 'Droid Sans Mono', 'Liberation Mono', Menlo, Courier, monospace",
```
[the remark this answers is not in the document]
Comment (nloureiro) [reply src/@chakra-ui/gatsby-plugin/theme.ts:40]: ok, agree
Thank you

**Comment by pettinarip** [comment]
> My take it's that once re touch the monospace family here, we might as well use this to add this font and QA all at the same time

Agree. Lets add it now and test all the font changes in this PR. @TylerAPfledderer how do you see doing this?

**Comment by TylerAPfledderer** [comment]
> > My take it's that once re touch the monospace family here, we might as well use this to add this font and QA all at the same time
> 
> Agree. Lets add it now and test all the font changes in this PR. @TylerAPfledderer how do you see doing this?

I would include the new font the same as with Inter, having a CSS file in the `./static` directory to expose it to Storybook, or one consolidated CSS file of all the applied `@font-face`'s.

Are you asking about testing as well?

**Comment by nloureiro** [comment]
I think he is referring to add the new font to this PR, and I will Re-test all the pages to see if all is good.

right?

**Comment by pettinarip** [comment]
Correct @nloureiro and @TylerAPfledderer.

Lets add the IBM font in this PR and then, Nuno and I, re-test everything again with the two fonts.

> @pettinarip I would include the new font the same as with Inter, having a CSS file in the ./static directory to expose it to Storybook, or one consolidated CSS file of all the applied @font-face's.

I think it will be best to have it in its own file, `static/fonts/ibm-plex-font-face.css` + a new entry in the preload scripts of `gatsby-ssr`

**Comment by TylerAPfledderer** [comment]
> I think it will be best to have it in its own file, static/fonts/ibm-plex-font-face.css + a new entry in the preload scripts of gatsby-ssr

@pettinarip on it!

**Comment by TylerAPfledderer** [comment]
@nloureiro @pettinarip this should be good to check with the new font!

**Review by nloureiro — APPROVED** [review]
I've checked a lot of pages in various OSs and all looks good!

**Review by pettinarip — APPROVED** [review]
Wohooo! lgtm thanks @TylerAPfledderer 

cc @nloureiro merging this

**Pull request merged**